bayesian/license_extractor.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""License extractor utility functions."""
+
+from .default_config import LIC_SYNONYMS_URL
+from requests import get
+from collections import defaultdict
+from lru import lru_cache_function
+from flask import current_app
+from datetime import datetime as dt
+
+
+# caching response for 24 hours
+@lru_cache_function(max_size=2048, expiration=60 * 60 * 24)
+def get_license_synonyms():
+    """Fetch all the license sysnonyms from license anlysis github repo."""
+    resp = get(LIC_SYNONYMS_URL)
+    if resp.status_code == 200:
+        current_app.logger.info(
+            "{} Succefully fetched license synonyms".format(dt.now()))
+        return resp.json()
+    else:
+        current_app.logger.error("{tm} Unable to fetch license synonyms, STATUS_CODE:{cd}".format(
+            tm=dt.now(), cd=resp.status_code))
+        return {}
+
+
+def extract_licenses(license_files):
+    """Extract license name from the given license files."""
+    lic_syn = get_license_synonyms()
+    response = dict()
+    if lic_syn:
+        for f_no, _file in enumerate(license_files, 1):
+            license_key = None
+            try:
+                # remove all the unnecessary chars ',', \n, space
+                content = _file.read().decode('utf-8')
```

Comment:
In Python 3 `read()` returns `str`, i.e. decoded string, so the `decode('utf-8')` won't actually work